field/src/helpers.rs

```
@@ -54,8 +54,29 @@ where
     F: Field,
     Y: Iterator<Item = F>,
 {
-    // TODO: Use PackedField
-    x.iter_mut().zip(y).for_each(|(x_i, y_i)| *x_i += y_i * s);
+    let y_vec: Vec<F> = y.collect();
+    let len = y_vec.len().min(x.len());
```

It would be nice to try and do this without needing the collect.

Also I think we can assume that `x.len() <= y.len()`. (These should be equal but at the very least `y` needs to be longer or this should throw an error).